src/ResourceManager/Network/Commands.Network/ApplicationGateway/SslCertificate/AzureApplicationGatewaySslCertificateBase.cs

```diff
@@ -29,24 +30,50 @@ public class AzureApplicationGatewaySslCertificateBase : NetworkBaseCmdlet
         public string Name { get; set; }
 
         [Parameter(
-               Mandatory = true,
+               Mandatory = false,
                HelpMessage = "Path of certificate PFX file")]
         [ValidateNotNullOrEmpty]
         public string CertificateFile { get; set; }
 
         [Parameter(
-               Mandatory = true,
+               Mandatory = false,
                HelpMessage = "Certificate password")]
         [ValidateNotNullOrEmpty]
         public SecureString Password { get; set; }
 
+        [Parameter(
+               Mandatory = false,
+               HelpMessage = "SecretId (uri) of the KeyVault Secret. Use this option when a specific version of secret needs to be used.")]
+        [ValidateNotNullOrEmpty]
+        public string KeyVaultSecretId { get; set; }
+
+        [Parameter(
+               Mandatory = false,
+               HelpMessage = "KeyVault Secret Object. This option will use a version-less url for the certificate.")]
+        [ValidateNotNullOrEmpty]
+        public PSKeyVaultSecret KeyVaultSecret { get; set; }
```

Comment:
@akshaysngupta we can directly reference a type from another module -- this will cause a lot of issues and a coupling between the KeyVault and Network modules. The best thing we can do is have this `PSKeyVaultSecret` class inherit from a base class in common code that both KeyVault and Network have access to, but this will require a good amount of additional work. I think our best option is to avoid adding the `-KeyVaultSecret` parameter and just add the `-KeyVaultSecretId` parameter, which still exposes the functionality to the users.